react/features/e2ee/middleware.js

```diff
@@ -116,6 +130,15 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
             dispatch(setEveryoneSupportE2EE(false));
         }
 
+        if (isMaxModeReached(getState)) {
```

Comment:
Can you please try to unify the 3 blocks that look similar into one helper `updateMaxMode` (sample name).